core/src/main/java/gnu/trove/THashMap.java

```diff
@@ -743,6 +743,32 @@ public V setValue(V o) {
                                 // setValue is called again
             return prev;
         }
+
+
+        @Override
+        public final boolean equals(Object o) {
+            if (o == this) {
+                return true;
+            }
+            if (o instanceof Map.Entry) {
+                Map.Entry<?, ?> otherEntry = (Map.Entry<?, ?>) o;
+                Object otherKey = otherEntry.getKey();
+                Object otherValue = otherEntry.getValue();
+                //noinspection unchecked
+                if (!(key == otherKey || key != null && _hashingStrategy.equals(key, (K) otherKey))) {
```

Comment:
This implementation violates the `Entry` spec. It's written in the spec how exactly keys and values should be compared:

> Compares the specified object with this entry for equality. Returns true if the given object is also a map entry and the two entries represent the same mapping. More formally, two entries e1 and e2 represent the same mapping if
> ```java
>               (e1.getKey()==null ?
>                e2.getKey()==null : e1.getKey().equals(e2.getKey()))  &&
>               (e1.getValue()==null ?
>                e2.getValue()==null : e1.getValue().equals(e2.getValue()))
> ```
> This ensures that the equals method works properly across different implementations of the Map.Entry interface.

Moreover, this `equals` implementation violates the general contract for `Object.equals`. If compared with `Map.Entry` that comes from another Map, the hashingStrategy of this Map will be used, so it's possible to violate the symmetry of equals (a.equals(b) != b.equals(a)). Is it really necessary to use hashingStrategy here? 

We can compare hashingStrategy with custom comparator of the `TreeMap`. The custom comparator may be inconsistent with `equals`. Yet, `TreeMap.Entry` doesn't use the comparator to compare entries. Instead, it fulfills the `equals` contract. Also, `TreeMap.EntrySet` implementation delegates its equals to `AbstractSet` like you are doing, and it looks like everybody is fine with this. I suggest to reuse that experience here as well: ignore hashing strategy when comparing views.